plugins/org.obeonetwork.m2doc/src/org/obeonetwork/m2doc/parser/BodyParser.java

```diff
@@ -496,25 +520,115 @@ private Query parseQuery() throws DocumentParserException {
     }
 
     /**
-     * parses the options of an image tag.
+     * Parses all options of the given tag.
      * 
      * @param tag
      *            the tag to parse
+     * @param tokenType
+     *            the token type triggering the option retrieval.
+     * @param optionGroupRank
+     *            the rank of the option in the tag.
+     * @param optionValGroupRank
+     *            the rank of the option's value in the tag.
+     * @param image
+     *            the result image were to put error messages.
+     * @param optionPattern
+     *            the authorized pattern for the option.
      * @return the option map created.
+     * @throws DocumentParserException
+     *             if tag is syntactically invalid.
      */
-    Map<String, String> parseImageOptions(String tag) {
+    Map<String, String> parseOptions(String tag, TokenType tokenType, int optionGroupRank, int optionValGroupRank,

```

Comment:
This method should be extracted in a class OptionParser where the different phase of parsing are separated : 
 - parsing of the option key,
 - detection of the first '
 - parsing of the value.
These algorithm deserve to be designed in a clear fashion so as to avoid problems.